controllers/volumereplication_controller.go

```diff
@@ -188,10 +188,12 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 
 			return reconcile.Result{}, err
 		}
-		if err = r.addFinalizerToPVC(logger, pvc); err != nil {
-			logger.Error(err, "Failed to add PersistentVolumeClaim finalizer")
+		if pvc.GetDeletionTimestamp().IsZero() {
```

Comment:
if the PVC is getting deleted should we just return error at https://github.com/csi-addons/kubernetes-csi-addons/blob/main/controllers/replication.storage/volumereplication_controller.go#L200 and let the next reconcile fail?